src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -15277,10 +15276,6 @@ void Unit::ProcDamageAndSpellFor(bool isVictim, Unit* target, uint32 procFlag, u
                         takeCharges = true;
                     break;
                 case SPELL_AURA_MECHANIC_IMMUNITY:
-                    // Compare mechanic
```

Comment:
why?

Reply:
Same code as SPELL_AURA_MOD_MECHANIC_RESISTANCE, case slips into it now

Reply:
nope not the same here we also make takeCharges true

Reply:
Please recheck, SPELL_AURA_MOD_MECHANIC_RESISTANCE also sets makes takeCharges true

Reply:
ok nvm its github shit view i look in unit cpp through vim and i see all is ok can be resolved